app/models/host/base.rb

```diff
@@ -195,8 +195,15 @@ def set_interfaces(parser)
       # is saved to primary interface so we match it in updating code below
       if !self.managed? && self.primary_interface.mac.blank? && self.primary_interface.identifier.blank?
         identifier, values = parser.suggested_primary_interface(self)
-        self.primary_interface.mac = Net::Validations.normalize_mac(values[:macaddress]) if values.present?
         self.primary_interface.update_attribute(:identifier, identifier)
+        if values.present?
+          self.primary_interface.mac = Net::Validations.normalize_mac(values[:macaddress])
+          # bridge interfaces are not attached to parent interface so save would not be possible
+          if interface_class(identifier) != Nic::Bridge && values[:virtual].present? && values[:attached_to].present?
```

Comment:
another virtual interface type is bond and potentially alias, but this seems to only support vlan

also why did you changed the order of this and the `update_attribute` on line above?

Reply:
It should support all virtual types except bridge, I can add tests for this. As long as virtual flag is present I guess.

The order was not intentional, I just felt like I want to set the mac in `if values.present?` block for better readability. Shall I put it back? I don't see any failures so far.

Reply:
Yes please, since uodate_attribute triggers the save, I think the order matters